packages/actor-extract-links-solid-type-index/lib/ActorExtractLinksSolidTypeIndex.ts

```diff
@@ -144,6 +147,68 @@ export class ActorExtractLinksSolidTypeIndex extends ActorExtractLinks {
     return typeLinks;
   }
 
+  /**
+   * To fetch the domain of the predicate.
+   * @param predicateSubjects A dictionary of predicate and its subjects from the query.
+   * @param typeSubjects A dictionary of class type and its subjects from the query.
+   */
+  public async linkPredicateDomains(predicateSubjects: Record<string, RDF.Term>,
+    typeSubjects: Record<string, RDF.Term[]>): Promise<void> {
+    if (Object.keys(predicateSubjects).length > 0) {
+      const predicateDomainsRec = (await Promise.all(Object.keys(predicateSubjects)
+        .map(predicate => this.fetchPredicateDomains(predicate))))
+        // eslint-disable-next-line unicorn/prefer-object-from-entries
```

Comment:
This linter rule gives you a hint to simplify this code.
Have a look at `Object.fromEntries()`, it might help making this piece of code a bit more compact.